src/retrace/retrace.py

```diff
@@ -1078,7 +1078,7 @@ def get(self, key: Union[str, Path], maxlen: int = 268435456) -> Optional[str]:
             return None
 
         filename = self._get_file_path(key)
-        with open(filename, "r") as f:
+        with open(filename, "r", encoding='utf-8') as f:
```

Comment:
Should we also pass `errors` with `replace` or something other as the value? Just in case of corruption.

Reply:
idk, would it not be better to fail in case of corruption?

Reply:
Are we failing in a controlled manner or just giving the user an unhandled exception?

Reply:
It would be better to have things just work. I don’t know what the consequences would be of showing replacement characters in cases where something magically fails.

Reply:
Ok, I think you are right. I added the errors='replace' to the open().

Reply:
I like the addition of 'errors' handler.  In this case, I think errors='ignore' is better than errors='replace' though, unless we actually want garbage / non-ascii characters printed?

I tested both errors='ignore' and errors='replace' with an inserted 0xff character in one retrace_server log file and then using a web browser to dump the retrace_log (via manager interface).  As expected, 'ignore' does not print anything but 'replace' prints a garbage character at the end.  If you don't have an errors= then this gives a Traceback internally and "Internal server error" on the manager interface / browser.

Reply:
FWIW, reference to the handlers - https://docs.python.org/3/library/codecs.html#error-handlers

Reply:
If you wanted to get fancy there may be some merit in some sort of exception handling and printing to the httpd log the location of the bad bytes as would occur if there were no 'errors=' on the open, or a 'try .. except' clause:
~~~
error decoding file /cores/retrace/tasks/497308754/retrace_log: 'utf-8' codec can't decode byte 0xff in position 429: invalid start byte
~~~

The above may be overkill, but this is the 'get()' method so it handles a lot of files, though most should not have problems with non-ascii chars. In my experience decoding errors have only occurred when some other programs output is dumped into a file (in this case, wget for whatever reason put a non-ascii character in there, and in one other case 'crash' output was non-ascii).

Reply:
I really do not like loosing information :) Replacing a non ascii character with <?> at least tells us that there are non-ascii characters. If we just ignore them we will loose this information. Now, I do not think it is very valuable information in most cases, but it may be in some. Unless the replace method causes issues (IIUC it does not) I would go with that.

Reply:
> In my experience decoding errors have only occurred when some other programs output is dumped into a file (in this case, wget for whatever reason put a non-ascii character in there, and in one other case 'crash' output was non-ascii).

Bingo. Note that I intentionally did not dig to see if there are scenarios where we could get garbage without cosmic rays being involved, so it might very well be just a precaution, but it seems like a cheap way to prevent something like that without erroring out, so why not.